### PR TITLE
fix: support contract JSON ABI with callpaths

### LINKED ISSUE #1320: Support JSON ABI with callpaths
Consider the following Sway contract:
```
contract;

use std::address::Address;

abi Reproduction {
    fn example_fn(address: Address);
}

impl Reproduction for Contract {
    fn example_fn(address: Address) {}
}
```

A bare `forc build` will result in the following:
```
{
  "types": [
    {
      "typeId": 0,
      "type": "()",
      "components": [],
      "typeParameters": null
    },
    {
      "typeId": 1,
      "type": "b256",
      "components": null,
      "typeParameters": null
    },
    {
      "typeId": 2,
      "type": "struct Address",
      "components": [
        {
          "name": "value",
          "type": 1,
          "typeArguments": null
        }
      ],
      "typeParameters": null
    }
  ],
  "functions": [
    {
      "inputs": [
        {
          "name": "address",
          "type": 2,
          "typeArguments": null
        }
      ],
      "name": "example_fn",
      "output": {
        "name": "",
        "type": 0,
        "typeArguments": null
      },
      "attributes": null
    }
  ],
  "loggedTypes": [],
  "messagesTypes": [],
  "configurables": []
}
```

However, `forc build` also supports the following flag:
```
        --json-abi-with-callpaths
            Outputs json abi with callpaths instead of names for struct and enums
```

Using the flag results in the following ABI:
```
{
  "types": [
    {
      "typeId": 0,
      "type": "()",
      "components": [],
      "typeParameters": null
    },
    {
      "typeId": 1,
      "type": "b256",
      "components": null,
      "typeParameters": null
    },
    {
      "typeId": 2,
      "type": "struct std::address::Address",
      "components": [
        {
          "name": "value",
          "type": 1,
          "typeArguments": null
        }
      ],
      "typeParameters": null
    }
  ],
  "functions": [
    {
      "inputs": [
        {
          "name": "address",
          "type": 2,
          "typeArguments": null
        }
      ],
      "name": "example_fn",
      "output": {
        "name": "",
        "type": 0,
        "typeArguments": null
      },
      "attributes": null
    }
  ],
  "loggedTypes": [],
  "messagesTypes": [],
  "configurables": []
}
```

Note the difference between the type fields for type ID 2 on each of the ABIs.

The `#[indexer(...)]` macro will build with the first ABI, but it will not build with the second as we do not check for the path separator when creating the `Ident` from the ABI types. We should ensure that the second one leads to a successful build as well as it will soon become the default option for all JSON ABIs:

> the requirement came from actual users having issues, this is to become the default once fuels ts supports it
-- @segfault-magnet, said on Slack

### PR DESCRIPTION
**Pull request by deekerno**
Closes #1320.

### Description

Adds support for JSON ABIs that have callpaths in the `type` field.

### Testing steps

The `fuel-indexer-test` contract ABI has been re-generated with `forc build --json-abi-with-callpaths` and CI should pass.

#### Manual testing

On `develop`:
- Go into any of the example contracts and run `forc build` to get a regular JSON ABI. Attempt to build the corresponding indexer for that contract. It should succeed. 🟢 
-  Run `forc build --json-abi-with-callpaths` to generate a JSON ABI with callpaths included. Attempt to build the corresponding indexer for that contract. It should fail. 🔴 

On this branch:
- Go into any of the example contracts and run `forc build` to get a regular JSON ABI. Attempt to build the corresponding indexer for that contract. It should succeed. 🟢 
-  Run `forc build --json-abi-with-callpaths` to generate a JSON ABI with callpaths included. Attempt to build the corresponding indexer for that contract. It should also succeed. 🟢

### CODE REVIEW

**Review by ra0x3 — APPROVED** [review]
Will defer to @lostman

File: packages/fuel-indexer-macros/src/indexer.rs [hunk 1]
```diff
@@ -45,7 +45,12 @@ fn process_fn_items(
     let mut abi_dispatchers = Vec::new();
 
     let funcs = abi.clone().functions;
-    let abi_types = abi.clone().types;
+    let abi_types: Vec<TypeDeclaration> = abi
+        .clone()
+        .types
+        .iter()
+        .map(|t| strip_callpath_from_type_field(t.clone()))
```
Comment (lostman) [inline packages/fuel-indexer-macros/src/indexer.rs:52]:
```suggestion
        .into_iter()
        .map(strip_callpath_from_type_field)
```

**Review by lostman — APPROVED** [review]

File: packages/fuel-indexer-macros/src/indexer.rs (hunk 1 above)
Comment (lostman) [reply packages/fuel-indexer-macros/src/indexer.rs:52]: `abi` is already `clone()`d

Comment (deekerno) [reply packages/fuel-indexer-macros/src/indexer.rs:52]: I'll clean this up with a `clippy` fix PR later. 👌🏽

**Pull request merged**